src/components/menu/positioning/PositionedMenuContent.tsx

```diff
@@ -0,0 +1,105 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import { isEqual } from 'lodash';
+import {
+    getViewportDimensionsAndCoords,
+    getElementDimensions,
+    getElementDimensionsAndCoords,
+    calculateMenuPosition,
+    IMenuPosition
+} from './positioningCalculations';
+import { IMenuPositionConfig } from '../MenuSharedTypes';
+
+export interface IPositionedMenuContentProps extends IMenuPositionConfig {
+    topLevelMenu: boolean;
+    togglerEl: HTMLElement;
+    children: React.ReactNode;
+}
+
+export interface IPositionedMenuContentState {
+    menuPosition: IMenuPosition;
+}
+
+export default class PositionedMenuContent extends React.Component<
+    IPositionedMenuContentProps,
+    IPositionedMenuContentState
+> {
+    public state: IPositionedMenuContentState = {
+        menuPosition: {
+            left: 0,
+            top: 0
+        }
+    };
+
+    private menuEl: HTMLElement = null;
+
+    public componentDidMount() {
+        this.calculateMenuPosition();
+    }
+
+    public componentDidUpdate(prevProps: IPositionedMenuContentProps) {
+        if (
+            prevProps.alignment !== this.props.alignment ||
+            prevProps.spacing !== this.props.spacing ||
+            prevProps.offset !== this.props.offset ||
+            prevProps.topLevelMenu !== this.props.topLevelMenu ||
+            prevProps.togglerEl !== this.props.togglerEl ||
+            prevProps.children !== this.props.children
+        ) {
+            this.calculateMenuPosition();
+        }
+    }
+
+    public componentWillMount() {
+        window.addEventListener('resize', this.calculateMenuPosition);
+        window.addEventListener('scroll', this.calculateMenuPosition);
+    }
+
+    public componentWillUnmount() {
+        window.removeEventListener('resize', this.calculateMenuPosition);
+        window.removeEventListener('scroll', this.calculateMenuPosition);
+    }
+
+    public render() {
+        return (
+            <div
+                style={{
+                    position: 'absolute',
+                    ...this.state.menuPosition
+                }}
+                ref={this.setElMenu}
+            >
+                {this.props.children}
+            </div>
+        );
+    }
+
+    private setElMenu = (el: HTMLElement) => {
+        this.menuEl = el;
+    }
+
+    private calculateMenuPosition = () => {
+        if (!this.props.togglerEl) {
+            return;
+        }
+        if (!this.menuEl) {
+            return;
+        }
+
+        const menuPosition = calculateMenuPosition({
```

Comment:
This is a bit confusing that calculateMenuPosition !== this.calculateMenuPosition

Reply:
Renamed to `positionMenu`